enterprise/admin/forms.py

```diff
@@ -52,23 +53,37 @@ def clean_email(self):
         return email
 
 
-class EnterpriseCustomerForm(forms.ModelForm):
+class EnterpriseCustomerAdminForm(forms.ModelForm):
     """
-    A custom model form to convert a CharField to a TypedChoiceField.
+    Alternate form for the EnterpriseCustomer admin page.
 
-    A model form that converts a CharField to a TypedChoiceField if the choices
-    to display are accessible.
+    This form fetches catalog names and IDs from the course catalog API.
     """
+    class Meta:
+        model = EnterpriseCustomer
+        fields = "__all__"
 
     def __init__(self, *args, **kwargs):
         """
-        Convert SlugField to TypedChoiceField if choices can be accessed.
+        Initialize the form.
+
+        Substitute a ChoiceField in for the catalog field that would
+        normally be set up as a plain number entry field.
         """
-        super(EnterpriseCustomerForm, self).__init__(*args, **kwargs)
+        super(EnterpriseCustomerAdminForm, self).__init__(*args, **kwargs)
+        self.fields['catalog'] = forms.ChoiceField(choices=self.get_catalog_options())
```

Comment:
We allow creating an EnterpriseCustomer before a catalog exists for that customer to use, right? (The field has `null=True` so that seems to be the goal of this PR as well). If so, there was a bug in this PR - the line above should also include `required=False`, or else it's impossible to create an EC without selecting a catalog:
![screen shot 2016-11-24 at 10 53 19 pm](https://cloud.githubusercontent.com/assets/945577/20616772/7e3241f2-b299-11e6-9f33-de6dae42f15d.png)

The above error is very inconvenient on a devstack for someone who doesn't have the catalog service running.